src/main/groovy/io/openliberty/tools/gradle/tasks/DevTask.groovy

```diff
@@ -166,6 +166,14 @@ class DevTask extends AbstractServerTask {
         this.container = container;
     }
 
+    private File dockerfile;
+
+    @Option(option = 'dockerfile', description = 'Dev mode will build a docker image from the provided Dockerfile and start a container from the new image.')
+    void setDockerfile(String dockerfile) {
+        File tmp = new File(dockerfile);
+        this.dockerfile = new File(tmp.getCanonicalPath()); // ensures the dockerfile is defined with the full path - matches how maven behaves
```

Comment:
`getCanonicalPath()` could throw IOException, consider whether we should catch that and rethrow an exception with a more user-oriented message.

Reply:
Updated